File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        string result="";
        for (int i =0;i<s.size();i++){
            if(!result.empty() && 
            (result.back()-32==s[i]||
            result.back()+32==s[i])){
                result.pop_back();
            }
            else{
                result.push_back(s[i]);
            }
        }
        return result;
    }
};

File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        result = []

        for ch in s:
            if result and abs(ord(result[-1]) - ord(ch)) == 32:
                result.pop()
            else:
                result.append(ch)

        return "".join(result)